apis/r/R/TileDBObject.R

```diff
@@ -9,18 +9,24 @@ TileDBObject <- R6::R6Class(
   public = list(
     #' @description Create a new TileDB object. (lifecycle: experimental)
     #' @param uri URI for the TileDB object
-    #' @param verbose Print status messages
     #' @param platform_config Optional platform configuration
-    #' @param tiledbsoma_ctx optional SOMATileDBContext
-    initialize = function(uri, platform_config = NULL, tiledbsoma_ctx = NULL) {
-      calls <- vapply(
-        X = lapply(X = sys.calls(), FUN = as.character),
-        FUN = '[[',
-        FUN.VALUE = character(length = 1L),
-        1L
-      )
-      if ('TileDBObject$new' %in% calls) {
-        .NotYetImplemented()
+    #' @param tiledbsoma_ctx Optional SOMATileDBContext
+    #' @param mode One of "READ" or "WRITE"
+    #' @param internal_use_only Character value to signal 'permitted' call as
+    #' `new()` is considered internal and should not be called directly
+    initialize = function(uri, platform_config = NULL, tiledbsoma_ctx = NULL,
+                          mode = "READ", internal_use_only = NULL) {
+      ## calls <- vapply(
+      ##   X = lapply(X = sys.calls(), FUN = as.character),
+      ##   FUN = '[[',
+      ##   FUN.VALUE = character(length = 1L),
+      ##   1L
+      ## )
+      ## if ('TileDBObject$new' %in% calls) {
+      ##   .NotYetImplemented()
+      ## }
+      if (is.null(internal_use_only) || internal_use_only != "allowed_use") {
+        stop("Use of the new() method is discouraged, consider using a factor method", call. = FALSE)
```

Comment:
Problem is that the message is a base class used for DataFrame, Sparse and Dense Array. That's why I refrained from pointing to a given method: it will look bad for a given call.

Reply:
ok but i think they do need _some_ indication of where to look for 'ok, what is this factory method thing i need to use, and how do i spell it'?

Reply:
We could derive in each class, know the name there, construct the proper alternative and bark with that.  And of course also call the `super$initialize(...)` to have it initialize.

But that seems like big hammer for a somewhat small nail.   

Maybe compromise and use "as e.g. `SOMADataFrameOpen()`"  ?

Reply:
yes the "e.g." bit is all i want -- thanks @eddelbuettel !